### PR TITLE
Format stakingBalance as votingBalance + nominatingBalance

### PR DESCRIPTION
**Pull request by jacogr**
Calculate stakingBalance correctly, Closes https://github.com/polkadot-js/apps/issues/206

polkadot apps portal 2018-08-07 16-21-48

### CODE REVIEW

**Comment by jacogr** [comment]
Will update with PR comments as soon as this actually reaches npm - https://travis-ci.org/polkadot-js/api/builds/413194530 (RxJs 6.x update)

File: packages/app-staking/src/observables/validatingBalance.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+type Result = [Balance, [Array<Balance>]];
+
+export default function validatingBalance (api: RxApiInterface, address: string): Observable<Balance> {
+  return combineLatest(
+    votingBalance(api, address),
+    api.state
+      .getStorage(storage.staking.public.nominatorsFor, address)
+      .pipe(
+        switchMap((nominators = []) =>
```
[the remark this answers is not in the document]
Comment (jacogr) [reply packages/app-staking/src/observables/validatingBalance.ts]: Thought about it: the issue here is not actually the start - it is the actual first result. 

Unless it has nominators, the storage entry does not exist. In that case it returns (actuallly throws) and the result is undefined. So in that case - empty array.

File: packages/app-staking/src/observables/validatingBalances.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    )
+  ).pipe(
+    map((result: Array<Balance>): BalanceMap => {
+      return result.reduce((balances, balance) => {
+        balances[balance.address] = balance;
+
+        return balances;
+      }, {} as BalanceMap);
+    })
+  );
```
[the remark this answers is not in the document]
Comment (jacogr) [reply packages/app-staking/src/observables/validatingBalances.ts]: This round going to keep as-is - it is pretty standard across the board on these with combine & map.

File: packages/app-staking/src/observables/validatingBalance.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import BN from 'bn.js';
+import { combineLatest } from 'rxjs/observable/combineLatest';
+import { switchMap } from 'rxjs/operators/switchMap';
+import { toArray } from 'rxjs/operators/toArray';
+import { map } from 'rxjs/operators/map';
+import storage from '@polkadot/storage';
+import encodeAddress from '@polkadot/util-keyring/address/encode';
+
+import votingBalance from './votingBalance';
+import votingBalances from './/votingBalances';
```
Comment (ltfschoen) [inline packages/app-staking/src/observables/validatingBalance.ts]: should be `'./votingBalances';`

**Pull request merged**